unblob/handlers/archive/instar/instar_hd.py

```diff
@@ -0,0 +1,44 @@
+from pathlib import Path
+
+from structlog import get_logger
+
+from unblob.file_utils import File, iterate_file
+from unblob.handlers.archive.zip import ZIPHandler
+from unblob.models import Extractor, HexString
+
+logger = get_logger()
+
+LOCAL_FILE_HEADER_ZIP = b"\x50\x4b\x03\x04"
+LOCAL_FILE_HEADER_INSTAR = b"\x50\x4b\x03\x07"
+
+EOCD_ZIP = b"\x50\x4b\x05\x06"
+EOCD_INSTAR = b"\x50\x4b\x05\x09"
+
+CD_FILE_HEADER_ZIP = b"\x50\x4b\x01\x02"
+CD_FILE_HEADER_INSTAR = b"\x50\x4b\x01\x08"
+
+
+class InstarHDExtractor(Extractor):
+    def fix_headers(self, chunk: bytes) -> bytes:
+        return (
+            chunk.replace(LOCAL_FILE_HEADER_INSTAR, LOCAL_FILE_HEADER_ZIP)
+            .replace(EOCD_INSTAR, EOCD_ZIP)
+            .replace(CD_FILE_HEADER_INSTAR, CD_FILE_HEADER_ZIP)
+        )
+
+    def extract(self, inpath: Path, outdir: Path):
+        with File.from_path(inpath) as file:
+            outfile = outdir.joinpath(f"{inpath.name}.zip")
+            with open(outfile, "wb") as out_file:
+                for chunk in iterate_file(file, 0, inpath.stat().st_size):
+                    out_file.write(self.fix_headers(chunk))
```

Comment:
Are ZIP headers always aligned ? Is it possible that a chunk ends in the middle of a header you want to replace ? 

paging our zip expert @e3krisztian 📟